tests/core/contracts/test_contract_call_interface.py

```diff
@@ -346,3 +346,28 @@ def test_call_undeployed_contract(undeployed_math_contract, call):
     with pytest.raises(BadFunctionCallOutput) as exception_info:
         call(contract=undeployed_math_contract, contract_function='return13')
     assert expected_undeployed_call_error_message in str(exception_info.value)
+
+
+def test_function_does_not_exist(arrays_contract, call):
+    with pytest.raises(AttributeError):
+        arrays_contract.functions.iDoNotExist().call()
+
+
+def test_argument_count(arrays_contract, call):
+    with pytest.raises(ValueError) as e:
+        arrays_contract.functions.setBytes32Value().call()
+    message, = e.value.args
+    assert message == (
+        "Incorrect number of arguments supplied or "
```

Comment:
I think I would prefer a custom exception type over asserting a specific message.  Since we don't have a `ValidationError` defined in web3.py I think this would be an appropriate time to add one.